crates/codegen/parser/runtime/src/support/stream.rs

```diff
@@ -1,12 +1,21 @@
 use std::mem;
 use std::ops::Range;
 
+use crate::parse_error::ParseError;
+
 use super::super::text_index::TextIndex;
 
 pub struct Stream<'s> {
```

Comment:
Given that this is now more than a stream, it should be renamed to `ParserContext` - either including a `Stream` or having `trait Stream`. And wherever `&mut Stream` is used the param name changed as well.